extensions/applicationinsights-cfgsync-js/src/CfgSyncHelperFuncs.ts

```diff
@@ -39,3 +42,114 @@ export function replaceByNonOverrideCfg<T=IConfiguration & IConfig, T1=NonOverri
+//                                |--------------------------------------------------------------------------|
+//                                |                    | none        | disabled    | enabled     | force     |
+//                                | ------------------ | ----------- | ----------- | ----------- | --------- |
+// | User Mode, value = A || C    | none               | none        | disabled    | disabled    | enabled   |
+// (user Value = A)               | disabled           | disabled    | disabled    | disabled    | enabled   |
+// (SDK Defaults = C)             | enabled            | enabled     | disabled    | enabled     | enabled   |
+//                                | none(blockCdn)     | none        | none        | none        | none      |
+//                                | disabled(blockCdn) | disabled    | disabled    | disabled    | disabled  |
+//                                | enabled(blockCdn)  | enabled     | enabled     | enabled     | enabled   |
+
+//                                cdn Mode (cdn Value = B)
+//                   |-----------------------------------------------------------|
+//                   |                    | none    | disabled | enabled | force |
+//                   | ------------------ | --------| ---------| --------| ------|
+// | User Mode       | none               | A       | A        | A       | B     |
+// (user Value = A)  | disabled           | A       | A        | A       | B     |
+//                   | enabled            | A || B  | A        | A || B  | B     |
+//                   | none(blockCdn)     | A       | A        | A       | A     |
+//                   | disabled(blockCdn) | A       | A        | A       | A     |
+//                   | enabled(blockCdn)  | A       | A        | A       | A     |
+
+export function resolveCdnFeatureCfg(field: string, cdnCfg?: ICfgSyncCdnConfig, customOptInDetails?: IFeatureOptIn) {
+    // cdn conifg value
+    if (!cdnCfg || !cdnCfg.enabled) {
+        return null;
+    }
+ 
+    let featureOptIn= cdnCfg.featureOptIn || {};
+    let cdnFeature = featureOptIn[field] || {mode: CdnFeatureMode.none};
+    let cdnMode = cdnFeature.mode;
+    let cdnFeatureVal = cdnFeature.value;
+    let customOptIn = customOptInDetails || {};
+    let customFeature = customOptIn[field] || {mode: FeatureOptInMode.disable}; // default custom mode is disable
+    let customMode = customFeature.mode;
+    let customFeatureVal = customFeature.cfgValue;
+    let blockCdn = !!customFeature.blockCdnCfg;
+    const fld = "featureOptIn.";
+    let mField = fld + field + ".mode";
+    let vField = fld + field + ".cfgValue";
+
+    if (blockCdn) {
+        return {
+            [mField]: customMode,
+            [vField]: customFeatureVal
+        };
+    }
+
+    if (cdnMode === CdnFeatureMode.force) {
+        return {
+            [mField]: FeatureOptInMode.enable,
+            [vField]: cdnFeatureVal
```

Comment:
This is the first bit that caused me to create the new table, and looking at the `setValueByKey` if the value used in undefined it "will" replace any existing value with undefined, and while we are "forcing" the feature on we may not be supplying any cdn override values so this probably should be B || A.
`[vField]: cdnFeatureVal || customFeatureVal` // We should be able to use this shortcut once we define this as always an object map.